src/App.js

```diff
@@ -2,26 +2,45 @@ import React, { Component } from 'react';
 import './App.css';
 import movieData from './mock-data';
 import Nav from './components/Nav/Nav.js';
-import Hero from './components/Hero/Hero.js';
 import Movies from './components/Movies/Movies.js';
-import Carousel from './components/Carousel/Carousel.js'
-
+import Carousel from './components/Carousel/Carousel.js';
+import fakeMovieDetailData from './fake-movie-detail-data.js';
+import MovieDetail from './components/MovieDetail/MovieDetail.js';
 class App extends Component {
   constructor() {
     super();
     this.state = {
       movies: movieData.movies,
+      // movies: [],
+      selectedMovieDetails: {},
     }
   } 
 
+  displayMovieDetail = (id) => {
+    // fetch(id) 
+    // ... returns detail data
+    this.setState({ selectedMovieDetails: fakeMovieDetailData.movie });
+  }
+
+  resetHome = () => {
+    this.setState({ selectedMovieDetails: {} });
+  }
+
   render() {
     return (
       <div>
-        <Nav />
-        <main>
-          <Carousel movies={this.state.movies} />
-          <Movies movies={this.state.movies} />
-        </main>
+        <Nav resetHome={this.resetHome} />
+        {!this.state.selectedMovieDetails.id && 
+          <main className='home-page'>
+            <Carousel movies={this.state.movies} displayMovieDetail={this.displayMovieDetail}/>
```

Comment:
I like how the `displayMovieDetail` is passed to several components!  I think we still need to add the onclick event to the hero image

Reply:
Yep